pkg/envinfo/envinfo.go

```diff
@@ -231,8 +243,22 @@ func (esi *EnvSpecificInfo) EnvInfoFileExists() bool {
 	return esi.envinfoFileExists
 }
 
+var (
+	// Mandatory parameters in the environment file (env.yaml)
+	manParams = []string{
+		"name",
+		"namespace",
+	}
+)
+
 // DeleteConfiguration is used to delete environment specific info from local odo envinfo
 func (esi *EnvSpecificInfo) DeleteConfiguration(parameter string) error {
+	for _, manParam := range manParams {
+		if parameter == manParam {
+			return errors.Errorf("failed to unset parameter %s: parameter %s is mandatory parameter", parameter, parameter)
```

Comment:
I think we should have quotes around the parameter here, as cases like `failed to unset parameter name: parameter name is mandatory parameter` is a little unclear. 

We can quickly format a string with quotes around it with `%q`

Reply:
Nit: `failed to unset parameter <parameter>: %s is a mandatory parameter` is a little cleaner

Reply:
👍